dht.go

```diff
@@ -37,9 +37,11 @@ var log = logging.Logger("dht")
 // collect members of the routing table.
 const NumBootstrapQueries = 5
 
-// IpfsDHT is an implementation of Kademlia with S/Kademlia modifications.
+type IpfsDHT = Node
+
+// Node is an implementation of Kademlia with S/Kademlia modifications.
 // It is used to implement the base IpfsRouting module.
-type IpfsDHT struct {
+type Node struct {
```

Comment:
If we're keeping this struct public, I'd rather prefix it with DHT, and remove the DHT from the functions below. 
But actually I wonder if there's any use case to keep Node public; I suspect no. In that case I'd be happy with an unqualified `node` type demoted to private scope.

Reply:
the interface is wider than the routing interface; we would need a new interface for the DHT.

Reply:
To my knowledge, returning a private type allows consumers to access all its public methods, so you don't need to access them through an interface. In this case, the benefit is that it disallows constructing a struct by itself, and it doesn't leak the unqualified `Node` type to the public namespace.

BTW – I generally agree that there may be more interfaces to extract.

Reply:
You can't construct a variable of it's type though, so you can't assign it anywhere which makes it borderline useless.
Note that the interface of the DHT is wider than any extant interface we have.

Reply:
Right, you can assign it to a local variable with the `:=` operator, but to pass it around functions you need to declare an interface. I'd argue that's the correct way, though.

Reply:
I think it should be a public type, and that type should be returned explicitly by its constructors. See the Go idiom on returning concrete types and accepting interfaces.